catfs/mio/encrypt/format.go

```diff
@@ -32,17 +32,35 @@ import (
 	"crypto/cipher"
 	"crypto/hmac"
 	"encoding/binary"
+	"errors"
 	"fmt"
 	"io"
 
+	"github.com/sahib/brig/catfs/mio/compress"
 	chacha "golang.org/x/crypto/chacha20poly1305"
 	"golang.org/x/crypto/sha3"
 )
 
+// Flags indicate with what options a stream was encoded.
+// Some flags are not compatible to each other, see below.
+type Flags int32
+
 // Possible ciphers in Counter mode:
 const (
-	aeadCipherChaCha = iota
-	aeadCipherAES
+	// FlagEmpty is invalid
+	FlagEmpty = Flags(0)
+
+	// FlagEncryptAES256GCM indicates the stream was encrypted with AES256 in GCM mode.
+	// This should be fast on modern CPUs.
+	FlagEncryptAES256GCM = Flags(1) << iota
+
+	// FlagEncryptChaCha20 incidate that the stream was encrypted with ChaCha20.
+	// This can be a good choice if your CPU does not support the AES-NI instruction set.
+	FlagEncryptChaCha20
+
+	// FlagCompressedInside indicates that the encrypted data was also compressed.
+	// This can be used to decide at runtime what streaming is needed.
+	FlagCompressedInside
```

Comment:
Shall we leave room for possible new encryption algorithm, and push bit 1 more to the left for `FlagCompressedInside`, after all we have 32 bits to spare

Reply:
We could also add new encrypt flags after ``FlagCompressedInside``, but I agree it might be tidier to move it back a bit.